### PR TITLE
Send more debug information in hello command

### PR DESCRIPTION
**Pull request by jqr**
Allows for easier server-side debugging by sending the equivalient of an HTTP
Agent string.

Other notable change is handling a response that comes in over multiple packets. This basically had a FIXME comment.

### CODE REVIEW

File: lib/instrumental.js [hunk 1]
```diff
@@ -97,9 +101,17 @@ function instrumental_send(payload) {
     // Connetion Established. Lets go shopping
     state = "connected";
 
+    var cleanString = function(value) {
+      return String(value).replace(/\s+/, "_");
```
Comment (janxious) [inline lib/instrumental.js]: I feel like this should be `#gsub`.

`#replace` is… [different](http://apidock.com/ruby/String/replace).

File: lib/instrumental.js [hunk 2]
```diff
@@ -97,9 +101,17 @@ function instrumental_send(payload) {
     // Connetion Established. Lets go shopping
```
Comment (janxious) [inline lib/instrumental.js]: Spilling errar.

File: lib/instrumental.js (hunk 1 above)
Comment (netshade) [inline lib/instrumental.js]: Do you mean changing the `/\s+/` to `/\s+/g` ?

Comment (janxious) [inline lib/instrumental.js]: This is javascript. Ignore my suggestion.

**Comment by netshade** [comment]
Merged this from other branch, you can close this @jqr.

**Comment by netshade** [comment]
NM, autoclose confused me, how do I github

**Comment by janxious** [comment]
Pretty well, most of the time.

**Pull request merged**